File: ios/Model/Message/Incoming/DeviceInformationChangeMessage.swift
//
// Copyright © 2023 Nevis Security AG. All rights reserved.
//

import NevisMobileAuthentication

class DeviceInformationChangeMessage: OperationMessage {
	// MARK: Properties

	var name: String?
	var fcmRegistrationToken: String?
	var disablePushNotifications: Bool?
	var retryPolicy: TypedRetryPolicy?

	// MARK: Decodable

	enum CodingKeys: String, CodingKey {
		case name
		case fcmRegistrationToken
		case disablePushNotifications
		case retryPolicy
	}

	required init(from decoder: Decoder) throws {
		let container = try decoder.container(keyedBy: CodingKeys.self)
		self.name = try container.decodeIfPresent(String.self, forKey: .name)
		self.fcmRegistrationToken = try container.decodeIfPresent(String.self, forKey: .fcmRegistrationToken)
		self.disablePushNotifications = try container.decodeIfPresent(Bool.self, forKey: .disablePushNotifications)
		self.retryPolicy = try container.decodeIfPresent(TypedRetryPolicy.self, forKey: .retryPolicy)
		try super.init(from: decoder)
	}
}

File: android/src/main/java/ch/nevis/mobile/authentication/sdk/react/model/message/incoming/DeviceInformationChangeMessage.kt
/*
 * Copyright © 2023 Nevis Security AG. All rights reserved.
 */

package ch.nevis.mobile.authentication.sdk.react.model.message.incoming

import ch.nevis.mobile.authentication.sdk.react.model.serializer.RequestHeadersSerializer
import ch.nevis.mobile.authentication.sdk.react.model.serializer.RetryPolicySerializer
import ch.nevis.mobile.sdk.api.operation.RequestHeaders
import ch.nevis.mobile.sdk.api.operation.RetryPolicy
import kotlinx.serialization.Serializable

@Serializable
data class DeviceInformationChangeMessage(
    override val operationId: String,
    @Serializable(with = RequestHeadersSerializer::class)
    override val requestHeaders: RequestHeaders? = null,
    override val onSuccessProvided: Boolean,
    override val onErrorProvided: Boolean,
    val name: String? = null,
    val fcmRegistrationToken: String? = null,
    val disablePushNotifications: Boolean? = null,
    @Serializable(with = RetryPolicySerializer::class)
    val retryPolicy: RetryPolicy? = null
) : OperationMessage()
